### ISSUE TITLE
Sandbox Qubes daemons in dom0

### ISSUE THREAD

**Issue by DemiMarie** [QubesOS/qubes-issues#4232]
### Qubes OS version:

R4.0

### Affected component(s): 

dom0

---

### Steps to reproduce the behavior:

Observe that various Qubes daemons in dom0 run with more privileges than they need.

### Expected behavior:

All Qubes daemons should run with the minimal required privileges

### Actual behavior:

Qubes daemons run unsandboxed.

### General notes:

While being able to compromise (say) a GUI daemon currently means fatal compromise of dom0, there is no fundamental reason this must be the case.  By properly sandboxing Qubes’s daemons, we can ensure that breaking out from there to a full dom0 compromise requires finding a privilege-escalation bug in the Linux kernel, where the kernel’s attack surface has been greatly reduced by means of seccomp-bpf, SELinux, running as an unprivileged user, and other measures.  Furthermore, unlike the dom0 userland, the dom0 kernel is kept up to date.

---

**Comment by marmarek** [comment]
I agree. If we can have it at low cost, why not. But if that would require some significant effort, then probably not worth it.
